src/workspaces/useWorkspaces.test.ts

```diff
@@ -0,0 +1,62 @@
+import { DeepPartial } from '@terra-ui-packages/core-utils';
+import { workspaceProvider } from 'src/libs/ajax/workspaces/providers/WorkspaceProvider';
+import { reportError } from 'src/libs/error';
+import { workspacesStore } from 'src/libs/state';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+import { asMockedFn, renderHookInAct } from 'src/testing/test-utils';
+import { useWorkspaces } from 'src/workspaces/useWorkspaces';
+
+type WorkspaceProviderExports = typeof import('src/libs/ajax/workspaces/providers/WorkspaceProvider');
+jest.mock(
+  'src/libs/ajax/workspaces/providers/WorkspaceProvider',
+  (): WorkspaceProviderExports => ({
+    workspaceProvider: {
+      list: jest.fn(),
+    },
+  })
+);
+
+type ErrorExports = typeof import('src/libs/error');
+jest.mock(
+  'src/libs/error',
+  (): ErrorExports => ({
+    ...jest.requireActual('src/libs/error'),
+    reportError: jest.fn(),
+  })
+);
+
+describe('useWorkspaces hook', () => {
+  beforeEach(() => {
+    workspacesStore.set([]);
+  });
+  it('calls workspaces provider and retains state in workspacesStore', async () => {
+    // Arrange
+    asMockedFn(workspaceProvider.list).mockResolvedValue([
+      { workspace: { name: 'myWorkspace' } } satisfies DeepPartial<WorkspaceWrapper> as WorkspaceWrapper,
+    ]);
+
+    // Act
+    const hookRender1 = await renderHookInAct(useWorkspaces);
```

Comment:
Style note: I've been using this naming convention for `renderHook`.

```suggestion
    const { result: hookReturnRef } = await renderHookInAct(useWorkspaces);
```

Reply:
hmmm... I kinda like having the .result.current be near the act since that lines up with cases where I want to save multiple result.current values for later assertions...
```

    // Act
    const hookRender = await renderHookInAct(useWorkspaces);
    const hookResult1 = hookRender.result.current;

    // Assert
    expect(workspaceProvider.list).toBeCalledTimes(1);
    expect(hookResult1.workspaces).toEqual([]);
```

Reply:
The other style doesn't prevent that.

```
// Act
const { result: hookReturnRef } = await renderHookInAct(useWorkspaces);
const hookResult1 = hookReturnRef.current;

// Assert
expect(workspaceProvider.list).toBeCalledTimes(1);
expect(hookResult1.workspaces).toEqual([]);
```

Originally, I think we came up with that style to avoid something like:
```ts
const result = renderHook(...)
expect(result.result.current)...
```

And to clarify that the `result` field behaves like a ref.
https://github.com/DataBiosphere/terra-ui/pull/3468#discussion_r1006213031